library/std/src/sync/lazy_lock.rs

```diff
@@ -103,23 +151,25 @@ impl<T: Default> Default for LazyLock<T> {
 #[unstable(feature = "once_cell", issue = "74465")]
 impl<T: fmt::Debug, F> fmt::Debug for LazyLock<T, F> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("Lazy").field("cell", &self.cell).finish_non_exhaustive()
+        f.debug_tuple("LazyLock")
+            .field(match self.get() {
+                Some(v) => v,
+                None => &"Uninit",
+            })
+            .finish()
     }
 }
 
 // We never create a `&F` from a `&LazyLock<T, F>` so it is fine
 // to not impl `Sync` for `F`
-// we do create a `&mut Option<F>` in `force`, but this is
-// properly synchronized, so it only happens once
-// so it also does not contribute to this impl.
 #[unstable(feature = "once_cell", issue = "74465")]
-unsafe impl<T, F: Send> Sync for LazyLock<T, F> where OnceLock<T>: Sync {}
+unsafe impl<T: Sync + Send, F: Send> Sync for LazyLock<T, F> {}
 // auto-derived `Send` impl is OK.
 
 #[unstable(feature = "once_cell", issue = "74465")]
-impl<T, F: UnwindSafe> RefUnwindSafe for LazyLock<T, F> where OnceLock<T>: RefUnwindSafe {}
+impl<T: RefUnwindSafe + UnwindSafe, F: UnwindSafe> RefUnwindSafe for LazyLock<T, F> {}
 #[unstable(feature = "once_cell", issue = "74465")]
-impl<T, F: UnwindSafe> UnwindSafe for LazyLock<T, F> where OnceLock<T>: UnwindSafe {}
+impl<T: UnwindSafe, F: UnwindSafe> UnwindSafe for LazyLock<T, F> {}
```

Comment:
Looks like these bounds are equal to the previous set:

```rust
impl<T: Send> Send for OnceLock<T>
impl<T: Sync + Send> Sync for OnceLock<T>
impl<T: RefUnwindSafe + UnwindSafe> RefUnwindSafe for OnceLock<T>
impl<T: UnwindSafe> UnwindSafe for OnceLock<T>
```

Reply:
Yes, exactly. It seemed better to me to enforce them separately here instead of relying on the bounds on `OnceLock`, especially since it is no longer used internally.

Reply:
Yeah I just wanted to note this for when this gets reviewed by someone from libs.